status-im/status-react
# Allow toggling pfs in dev-mode

### PFS toggle
This commits allow users to toggle pfs in dev-mode (it is already used
for pairing and group chats).

Once enabled direct and public messages sent will only be received by
users running >= 0.9.32.

Also this does not guarantee PFS (both devices need to have it enabled
it), and there still some UX work to do to ensure that, but it is useful
for testing.

A warning is displayed explaining the limitations when enabling.

### Show popup when device is not targeted.

When someone is sending a pfs message to us but did not include our own
device, a pop up is shown propmting the user to connect with the user.

The reason for receiving messages that are not targeting our devices are
various:

1) The account was just recovered (which means it is a new installation
id)
2) More than 3 devices are in use (we only keep max 3 devices in sync)
3) The sender has used an old bundle which does not include the current
device

Eventually we will reduce the likelihood of this scenario happening, but
we can't dismiss it completely.

It's only enabled when PFS is enabled.

### Testing

1) Create account A1
2) Create account B1
3) Enable PFS on both devices and exchange a few messages
4) Recover account A2 (same seed as A1, different device)
5) Enable PFS on account A2
6) Send a message from B1 to A

You should see a pop up on device A2 (on device A1 the message should be received correctly).

If you click on Cancel, nothing will happen and if another message is sent you will see the same popup.
If you click on `Add to contacts`, the next message from B1 should be received by both A1 & A2.

status: ready
